### PR TITLE
Refactor/question set detail page

### PR DESCRIPTION
**Pull request by TomasNiemczyk98**
## Summary

1st iteration of question set detail page, some stuff is slightly off, something is not working, but hopefully a step in the right direction

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] Refactoring (non-breaking change to code and structure)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

https://user-images.githubusercontent.com/92929243/161975710-bc353fd2-402a-4946-bed1-04d77b4eef58.mp4

### CODE REVIEW

**Comment by petrspelos** [comment]
@TomasNiemczyk98 while the circle buttons currently have the incorrect color, the ones with a checkbox in them should have a dashed type. This is something AntDesign gives us for free. You think we could use that?

```
 <Button type="dashed" shape="circle" ... />
```

**Comment by TomasNiemczyk98** [comment]
> @TomasNiemczyk98 while the circle buttons currently have the incorrect color, the ones with a checkbox in them should have a dashed type. This is something AntDesign gives us for free. You think we could use that?
> 
> ```
>  <Button type="dashed" shape="circle" ... />
> ```

My bad I did not even realize that AntDesign's buttons can be customized in this way, I will change implementation according to this

File: src/domain/QuestionList/QuestionSetView.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        <div className={styles.questionSetView}>
+          <AverageDifficultyCircle
+            percent={Number(
+              (
+                average(
+                  list.questions
+                    ?.filter((question) => question.difficulty)
+                    .map((question) => question.difficulty ?? 0) ?? []
+                ) * 20
+              ).toFixed(2)
```
Comment (PlesnikJakub) [inline src/domain/QuestionList/QuestionSetView.tsx:45]: I think it's not ideal to compute the average difficulty in this place.  I propose we rework this to a separate function that will take an array of questions and returns average difficulty. That would create one place responsible for such logic and with future changes to the difficulty attribute (already planned), we will have a more resilient solution. 

Regarding where should be such function stored, I would not hesitate to put it into some sort of service.

File: src/services/arrayUtils.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export function arrayDistinct<T>(arr: T[]): T[] {
```
Comment (PlesnikJakub) [inline src/services/arrayUtils.ts:1]: Not sure is this is **service** or just a **helper** 🤔

Comment (PlesnikJakub) [reply src/services/arrayUtils.ts:1]: I see we have multiple utils inside the service folder, we should probably consider what is the correct location.

**Review by PlesnikJakub — CHANGES_REQUESTED** [review]
There is a couple of small comments that prevent this to be merged.

@TomasNiemczyk98 please let me know if you have the capacity for the requested changes. We will take over if you won't be able to make it.

**Review by PlesnikJakub — APPROVED** [review]
Thank you for your contribution and resolution of our comments, I think this PR is in good enough shape to be merged now 🚀

**Pull request merged**